pkg/controller/jobframework/tas_validation.go

```diff
@@ -27,17 +27,22 @@ import (
 )
 
 func ValidateTASPodSetRequest(replicaPath *field.Path, replicaMetadata *metav1.ObjectMeta) field.ErrorList {
-	var allErrs field.ErrorList
+	allErrs := field.ErrorList{}
+	annotationsPath := replicaPath.Child("annotations")
+
 	requiredValue, requiredFound := replicaMetadata.Annotations[kueuealpha.PodSetRequiredTopologyAnnotation]
 	preferredValue, preferredFound := replicaMetadata.Annotations[kueuealpha.PodSetPreferredTopologyAnnotation]
 	_, unconstrainedFound := replicaMetadata.Annotations[kueuealpha.PodSetUnconstrainedTopologyAnnotation]
-	annotationFoundCount := 0
-	for _, found := range []bool{requiredFound, preferredFound, unconstrainedFound} {
-		if found {
-			annotationFoundCount++
+
+	// validate no more than 1 annotation
+	asInt := func(b bool) int {
+		if b {
+			return 1
+		} else {
+			return 0
 		}
```

Comment:
```suggestion
		}
		return 0
```

nit: This `else` statement does not provide any capability.

Reply:
Done.